src/main/resources/api/service_openapi.yaml

```diff
@@ -191,6 +194,228 @@ paths:
         '500':
           $ref: '#/components/responses/ServerError'
 
+  /api/workspaces/v1/{id}/resources/references/datarepo/snapshot:
+    parameters:
+    - $ref: '#/components/parameters/Id'
+    post:
+      summary: Create a new data repo snapshot reference in a workspace.
+      operationId: createDataRepoSnapshotReference
+      tags: [DataRepoReference]
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateDataRepoSnapshotReferenceRequestBody'
+      responses:
+        '200':
+          $ref: '#/components/responses/DataRepoSnapshotReferenceResponse'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/ServerError'
+
+  /api/workspaces/v1/{id}/resources/references/datarepo/snapshot/{referenceId}:
```

Comment:
I think we should discuss what level we should provide delete, get and enumerate support.
I am sure we need a top-level GET by name and id and enumerate across all resource types.
I am not sure we need per-type GET. The pro is, it is a simpler result. The con is, it is lots of endpoints that might never be used.
I do not think we need per-type DELETE.
I think we need per-type PATCH for controlled resources. Maybe for references, it is a wholesale replacement? A PUT?